CDS/js/test/youtube/TestYoutubeLib.js

```diff
@@ -0,0 +1,212 @@
+var chai = require('chai');
+var assert = chai.assert;
+var  chaiAsPromised = require('chai-as-promised');
+var sinon = require('sinon');
+
+chai.use(chaiAsPromised);
+
+var fs = require('fs');
+var sinon = require('sinon');
+var youtubeUpload = require('../../youtube/youtube-upload-lib');
+var youtubeAuth = require('../../youtube/youtube-auth');
+var googleapis = require('googleapis');
+var dataStore = require('../../Datastore');
+
+
+describe('YoutubeUpload', () => {
+    var stringSubstituteStub;
+
+    beforeEach(() => {
+        stringSubstituteStub = sinon.stub(dataStore, 'substituteStrings', (connection, val) => {
+            return new Promise((fulfill) => {
+                fulfill(val);
+            });
+        });
+
+    });
+
+    afterEach(() => {
+        stringSubstituteStub.restore();
+    });
+
+    after(() => {
+        delete process.env.cf_datastore_location;
+    });
+
+    describe('#getYoutubeData', () => {
+        var metadataStub, readStub;
+
+        beforeEach(() => {
+            process.env.cf_media_file = './media';
+            process.env.owner_account = 'account';
+            process.env.owner_channel = 'channel';
+            metadataStub = sinon.stub(youtubeUpload, 'getMetadata').returns(new Promise((fulfill, reject) => {
+                fulfill({});
+            }));
+            readStub = sinon.stub(fs, 'createReadStream');
+
+        });
+
+        afterEach(() => {
+            delete process.env.media_path;
+            delete process.env.owner_account;
+            delete process.env.owner_channel;
+            metadataStub.restore();
+            readStub.restore();
+        });
+
+        it('should parse youtube data correctly when all variables exist', () => {
+
+            return youtubeUpload.getYoutubeData(null)
+            .then((data) => {
+                assert.equal(data.part, 'snippet,status');
+                assert.ok(data.resource);
+                assert.ok(data.media);
+
+                assert.equal(data.onBehalfOfContentOwner, 'account');
+                assert.equal(data.onBehalfOfContentOwnerChannel, 'channel');
+                assert.equal(data.uploadType, 'multipart');
+                sinon.assert.calledOnce(stringSubstituteStub);
+                return;
+            });
+        });
+
+        it('should raise an exception if no media path provided', () => {
+
+            delete process.env.cf_media_file;
+
+            return assert.isRejected(youtubeUpload.getYoutubeData(), 'Cannot upload to youtube: missing media file path. Make sure that media has been specified in the route');
+
+        });
+
+        it('should raise an exception if channel is specified without an account owner', () => {
+            delete process.env.owner_account;
+
+            assert.isRejected(youtubeUpload.getYoutubeData(), 'Cannot upload to youtube: missing account owner');
+
+        });
+
+        it('should include paramters if account owner are both missing', () => {
```

Comment:
minor - typo in parameters and grammatically, this doesn't make sense as "both" is used when only one thing is listed.